app/src/main/java/com/randomappsinc/studentpicker/home/MainActivity.java

```diff
@@ -271,34 +274,14 @@ public void onRequestPermissionsResult(
         }
     }
 
-    private void showGoogleSpeechDialog() {
-        Intent intent = new Intent(RecognizerIntent.ACTION_RECOGNIZE_SPEECH);
-        intent.putExtra(RecognizerIntent.EXTRA_LANGUAGE_MODEL, RecognizerIntent.LANGUAGE_MODEL_FREE_FORM);
-        intent.putExtra(RecognizerIntent.EXTRA_LANGUAGE, Locale.getDefault());
-        intent.putExtra(RecognizerIntent.EXTRA_PROMPT, getString(R.string.list_name_input_speech_message));
-        try {
-            startActivityForResult(intent, SPEECH_REQUEST_CODE);
-        } catch (ActivityNotFoundException exception) {
-            UIUtils.showLongToast(R.string.speech_not_supported, this);
-        }
-    }
-
     @Override
     protected void onActivityResult(int requestCode, int resultCode, Intent data) {
         super.onActivityResult(requestCode, resultCode, data);
         switch (requestCode) {
             case SPEECH_REQUEST_CODE:
-                if (resultCode != RESULT_OK || data == null) {
-                    return;
-                }
-
-                List<String> result = data.getStringArrayListExtra(RecognizerIntent.EXTRA_RESULTS);
-                if (result == null || result.isEmpty()) {
-                    UIUtils.showLongToast(R.string.speech_unrecognized, this);
-                    return;
-                }
-                String searchInput = result.get(0);
-                newListInput.setText(searchInput);
+                String searchInput = SpeechUtil.processSpeechResult(resultCode, data, this);
+                if (searchInput != null)
```

Comment:
nit: Please add the brackets for consistency with the rest of the codebase.
```suggestion
                if (searchInput != null) {
```